.circleci/config.yml

```diff
@@ -174,285 +231,175 @@ jobs:
-          command: mkdir $DIST_DIR
-      - run:
-          name: Untar redhat
-          command: |
-            tar xavf redhat.tar.gz -C build
-      - run:
-          name: Untar orig
-          command: |
-            tar xavf varnish*.tar.gz -C build --strip 1
-      - run:
-          name: Build Packages
-          command: |
-            set -e
-            set -u
 
-            if [ << parameters.release >> = 8 ]; then
-                yum config-manager --set-enabled PowerTools
-            fi
-            # use python3
-            sed -i '1 i\%global __python %{__python3}' "$DIST_DIR"/redhat/varnish.spec
-            if [ -e .is_weekly ]; then
-                WEEKLY='.weekly'
-            else
-                WEEKLY=
-            fi
-            VERSION=$("$DIST_DIR"/configure --version | awk 'NR == 1 {print $NF}')$WEEKLY
-
-            cp -r -L "$DIST_DIR"/redhat/* "$DIST_DIR"/
-            tar zcf "$DIST_DIR.tgz" --exclude "$DIST_DIR/redhat" "$DIST_DIR"/
-
-            RPMVERSION="$VERSION"
-
-            RESULT_DIR="rpms"
-            CUR_DIR="$(pwd)"
-
-            rpmbuild() {
-            	command rpmbuild \
-            		--define "_smp_mflags -j10" \
-                	--define "_sourcedir $CUR_DIR" \
-                	--define "_srcrpmdir $CUR_DIR/${RESULT_DIR}" \
-                	--define "_rpmdir $CUR_DIR/${RESULT_DIR}" \
-            		--define "versiontag ${RPMVERSION}" \
-            		--define "releasetag 0.0" \
-            		--define "srcname $DIST_DIR" \
-            		--define "nocheck 1" \
-            		"$@"
-            }
-            yum-builddep -y "$DIST_DIR"/redhat/varnish.spec
-            rpmbuild -bs "$DIST_DIR"/redhat/varnish.spec
-            rpmbuild --rebuild "$RESULT_DIR"/varnish-*.src.rpm
-      - run:
-          name: Prepare the packages for storage
-          command: |
-            mkdir -p packages/<< parameters.dist >>/<< parameters.release >>
-            mv rpms/*/*.rpm packages/<< parameters.dist >>/<< parameters.release >>
-      - persist_to_workspace:
-          root: .
-          paths:
-            - packages/<< parameters.dist >>/<< parameters.release >>/*.rpm
-      - store_artifacts:
-          path: packages/<< parameters.dist >>/<< parameters.release >>
-          destination: ./
   collect_packages:
     docker:
       - image: centos:7
     steps:
       - attach_workspace:
           at: ~/project
+      - run: ls -la ~/project/
       - run:
           name: Tar the packages
           command: |
-              tar cvzf packages.tar.gz packages 
+              tar cvzf packages.tar.gz packages
       - store_artifacts:
           destination: packages.tar.gz
           path: packages.tar.gz
-pkg_req: &pkg_req
-  requires:
-    - dist
-    - tar_pkg_tools
+
 
 workflows:
   version: 2
   build:
     jobs:
       - dist
       - tar_pkg_tools
-      - build_debs:
-          name: build_debian_stretch
-          dist: debian
-          release: stretch
+      - package:
```

Comment:
that whole list is going to be condensed with the newly added matrix support. You can have a look, or we can do that later on

Reply:
I am not sure this will work in our case.
The new `matrix` support is a Cartesian product of all parameters. 
So it is going to look something like:
```
      - package:
          matrix:
            parameters:
              arch: ["aarch64", "x64"]
              dist: ["ubuntu", "debian", "centos", "alpine"]
              release: ["bionic", "xenial", "bullseye", "buster", "stretch", "7", "8", "3"]
              ext: ["deb", "rpm", "apk"]
              image: ["<< parameters.image_prefix >><< parameters.dist >>/<< parameters.release >>:<< parameters.version >>"]
              image_prefix: ["arm64v8/", ""]
              <<: *pkg_req
            exclude:
              - dist: ["ubuntu", "debian"]
                ext: ["rpm", "apk"]
              - dist: ["centos"]
                ext: ["deb", "apk"]
              - dist: ["alpine"]
                ext: ["rpm", "deb"]
              - arch: "aarch64"
                image_prefix: ""
              - arch: "x64"
                image_prefix: "arm64v8/"
```
And I didn't add exclusions for ubuntu+debian versions and vise versa.
IMO it is becoming more complicated with all the exclusions.
Also I needed to give up on the `-slim` versions of the Debian Docker images.

Reply:
arf, I'm now realizing that images are going to be an issue. You're right, let's keep it out of this PR